core/services/ardupilot_manager/flight_controller_detector/Detector.py

```diff
@@ -65,9 +95,14 @@ def detect() -> List[Tuple[FlightControllerType, str]]:
         """
         available = []
         if Detector._is_root():
-            result, path = Detector.detect_navigator()
+            # We should detect R4 first since it shares some sensors as R3
+            result, path = Detector.detect_navigator_r4()
+            if result:
+                available.append((FlightControllerType.NavigatorR4, path))
+
+            result, path = Detector.detect_navigator_r3()
             if result:
-                available.append((FlightControllerType.Navigator, path))
+                available.append((FlightControllerType.NavigatorR3, path))
```

Comment:
As discussed outside, this is making the manager detect a R4 as both R4 and R3, and choosing R3 since it's first on the FlightControllerType enum.